networking/ciao-cnci-agent/scripts/generate_cnci_cloud_image.sh

```diff
@@ -63,19 +63,18 @@ fi
 
 echo -e "\nMounting image: $image"
 tmpdir=$(mktemp -d)
-sudo modprobe nbd max_part=63
-sudo qemu-nbd --format=raw -c /dev/nbd0 "$image" 
 
 #it can take some time for the device to get created
 retry=0
+loop=''
 until [ $retry -ge 3 ]
 do
-	sudo udevadm settle
-	sudo partprobe /dev/nbd0
-	sudo mount /dev/nbd0p$partition "$tmpdir" && break
-	let retry=retry+1
-	echo "Mount failed, retrying $retry"
-	sleep 10
+    if [ "$loop" == "" ]; then
+	loop=`sudo losetup -f --show -P $image`
+    fi
+    sudo mount ${loop}p$partition "$tmpdir" && break
```

Comment:
The udevadm settle shouldn't needed, nor the retry-loop.  But there's a race between finding a free loopdev, storing it in 'loop'.  Kpartx encapsulates logic to find and use a free loopdev, eg:

kpartx -sav $image
(parse the output to find which loopN was selected)
mount ${loop}p$partition "$tmpdir"  && break
echo "fatal/non-recoverable error got in the way"